sunpy/net/vso/vso.py

```diff
@@ -710,17 +710,48 @@ def create_getdatarequest(self, maps, methods, info=None):
         fileids and methods, """
         if info is None:
             info = {}
+        if 'JSOC' in maps:
+            dris = self._separate_hmi(maps)
+        else:
+            dris = [self.make('DataRequestItem', provider=k, fileiditem__fileid=[v])
+                    for k, v in iteritems(maps)]
 
         return self.make(
             'VSOGetDataRequest',
             request__method__methodtype=methods,
             request__info=info,
-            request__datacontainer__datarequestitem=[
-                self.make('DataRequestItem', provider=k, fileiditem__fileid=[v])
-                for k, v in iteritems(maps)
-            ]
+            request__datacontainer__datarequestitem=dris
         )
 
+    def _separate_hmi(self, maps_dict):
+        """
+        This function parses the separate hmi data series from the
+        maps_dict(jsoc) list, pulls any hmi data it finds into separate lists
+        and deletes the hmi data from the original jsoc list.  It then continues
+        as normal to construct the soap message.
+        """
+        jsoc_list = maps_dict.get('JSOC', [])
+        listdict = defaultdict(lambda: [])
+        s_match = re.compile('^.*?:')
```

Comment:
An example hmi fileid has the form: `hmi__m_45s:13445764:13445764` where the series designation terminates at the first colon.  Adding the `?` to the regex makes the expression non-greedy, so that it matches only up to the first colon: `hmi__m_45s:` 

Without the `?` the expression also gobbles up the first number, `hmi__m_45s:13445764:` which would later on cause every unique fileid to get packaged in its own datarequestitem, instead of packaging them by series.

Reply:
This one is for @sudozer !

Reply:
For reference, a few fileids from the same time but different series look like:
`hmi__m_45s:13445764:13445764`
`hmi__v_45s:13445764:1344576`
`hmi__ic_45s:13445764:13445764`
So you can see how matching to the first colon is the most straightforward way of discriminating between the different series, which is the core purpose of _separate_hmi.  I don't think wrapping each individual hmi fileid in a datarequestitem would have catastrophic consequences, it may make the processing on the vso side a little slower though.